### ISSUE TITLE
Export specific quantity

### ISSUE THREAD

**Issue by TheChance** [AppliedEnergistics/Applied-Energistics-2#1720]
After much back-and-forth in #FTB, and an hour or so of experimenting, I've concluded that there's really no good way to export a specific quantity of items from an ME network. The best we can do is to keep a buffer chest stocked, and try to time a redstone pulse to empty it of more-or-less the correct number of units (or use Thermal Expansion ducts to pull the desired quantity on pulse, but come on).

In addition to being clunky, this solution has the unpleasant side-effect of removing more items from storage than necessary.

I'd like to be able to respond to a redstone signal by exporting a configurable quantity, or perhaps to be able to configure an export quantity and rely on pulses to restrict the number of exports.

Use case: I'm trying to set up an automated TiCon room, and through crossmod goodness, it's trivial to send a pulse when a molten metal tank falls below a certain level. However, there's no practical way (at least none that I can find) to have the ME network respond by exporting a specific number of ingots to the smeltery, which seems like it should be trivial.

**Comment by yueh** [comment]
The main issue here is that you want to monitor a fluid, which is not supported by AE2.

Keeping an exact stack of items at hand by monitoring an inventory is already possible and does not need the option to export a specific amount. Just not with fluids (directly).

An option would also make acceleration cards obsolete as these define how many items per tick are exported. Just rapidly pulsing it, will make them work way faster compared to cards.

With some redstone logic, you can even export a precise amount directly into the smeltery without any buffer. It might just either be fast enough (1 item/tick) or not precise enough (only multiples of 1, 8, 32, 64 or 96 items)

In relation to #16, I was thinking about adding a way to let it scale on the redstone strength and use it as percentage.
This will not make the cards obsolete, but still allow a bit more control about the amount. Combined with pulsing the bus it can be pretty precise.
Of course it has some drawbacks as it requires a card, so it will be limited to 3 acceleration cards max ( 1 stack/tick) and the redstone strength is limited to 15 values. Which comes down to a precision of 0.5 items per level with 1 card, about 2 with 2 cards and 4 with 3. 
But I would consider losing some precision with a higher speed a fair tradeoff.

To summarize, most likely not happening.

**Comment by TheChance** [comment]
I would regard the fluid monitoring as irrelevant, as I'm crossing mods to do that anyway. I can accomplish the same as described above using a configured ME Interface, TE ducts, and a pulse former + repeater.

It just seems like such a trivial task, to require three mods and a complex circuit: "When I supply a redstone signal, output exactly <x> qty of <y> item". I respect your thinking, but this is not the type of mechanism I want to be putting tons of time, thought or resources into. This should be a brainless component of a complex system, such as what I'm discussing with fluid monitoring.

**Comment by yueh** [comment]
On a second thought, I do not even see a need for scaling the buses or a complex redstone circuit at all.

AE2 can already export a very specific amount of items on a single redstone pulse, even something like 37 items of X and 26 of Y. While simultaneously being trivial to use.

**Comment by TheChance** [comment]
Really? That's fantastic, what am I missing? Neither I nor anyone in #FTB had a clue. It was my understanding that a configured interface still needed to be pumped from using some other mod.

**Comment by TheChance** [comment]
I have been scouring the internet and I honestly can't figure out what you're referring to. I'm not trying to be combative here; I'd really appreciate some insight, as I'm working on a rather large-scale build, and if AE2 *is* capable of handling this for me from one end to the other, I'll be able to compact it dramatically.

I realize that an ME interface can ready itself for another mod (or another ME network) to *pull* a specific quantity on pulse, but I haven't been able to find a way to get it to *push* on pulse, and neither has anyone in #FTB, any of the several times I've asked.

I must have rotated my Google terms around 20 different ways, and the closest thing I've found was a forum thread dating back to AE2's release, wherein someone suggested exactly what I'd have suggested - a "quantity card" for the export bus.

**Comment by yueh** [comment]
The main point here is that this is a bugtracker, so we will not solve build issues (here).
Some people might actually use it to report bugs, but still want to find their own solutions and not just read about existing ones.

All I will say, is that you probably need to reconsider your build. Trying to use a sledgehammer as as a pair of screwdriver while actually needing some tweezers is not our issue. There are more than just one ways to solve your problem with AE2. Some might be easier while others are more complex and we will not prevent you from using the complex solution, but also will not make them any easier.

**Comment by TheChance** [comment]
Realizing that it was written by Algorithm, are you aware that the FAQ at ae-mod.info instructs us to post feature suggestions here?

That's what I thought I was posting, a feature suggestion.

You say that the functionality I'm looking for already exists, fine. Literally -nobody in the FTB IRC channel- can figure out what it is, and you don't want to tell me.

So I'm a jerk for posting a feature suggestion where I was told, I'm a jerk for not being able to divine something that the rest of the community doesn't know either, and I'm a jerk because you're pretty sure my idea is stupid.

Thanks for all your help. I *sincerely* appreciate it. I've got my build working anyway, you know, I was just curious what mystery feature you were referring to.

**Comment by yueh** [comment]
Notice the *here*. I am pretty confident that you are able to find the AE IRC channel, if you are already using IRC. This here is simply not the right platform and way to clunky for it. 

I cannot speak about the quality of the FTB IRC channel. But if this in any case resembles the quality of the ftb subreddit, then I would not really be surprised, if half of the help there is simply wrong (which is just sad)

**Comment by Blue64** [comment]
Normally you're fairly helpful Yueh, but you didn't even reference a single item, block, panel, or interface, and now I'm also curious how you were going to explain exporting a set amount instead of importing it. (I'm finding these deeply nested in another issue xD sorry if it seems random)

**Comment by TheChance** [comment]
Yes, hence my attitude. It's possible to do this using a subnet, but there is nothing "trivial" about it.

Examine the material costs of a TE4 itemduct+servo on an interface, for instance, and compare it with the price of the sequence of buses and cables required to accomplish same via AE2.

It's no contest. Doing this with AE2 is needlessly complicated and expensive, when we're talking about a mod that provides so much ease-of-use in other respects. And there's already a component that does redstone pulses, but making it compatible with an interface would be overpowered/remove a challenge?

Naw. I needed a bank of 32 of this mechanism in survival. The current situation is underpowered. I solved the problem by using your crossmod component to let another mod do the logic - a mod which few would accuse of being "overpowered" in survival.

From my end, it seemed like Yueh read, "moving liquid," and decided I should piss off. Which is weird... getting frustrated with players who want to work around the limitations of your mod implies that there is a wrong way to sandbox.

Several months later, EC2 is buggy, this is still too expensive to pull off in AE2, and I'm still using Thermal Expansion ducts for exactly one job.

**Comment by Blue64** [comment]
I use Translocators, but I've never actually needed to use the liquid ones (Yet. Gonna need to when I build my Deep Ocean Biome Base Entrance, complete with Dense Oil Escalator and Elevator to the first Security System)

I like AE2 because it's expensive and complicated lol apart from being impressive, it makes everything run so smoothly, though I have noticed my laptop starts to Lag with a full ME Drive, but it's expected as it's a rather old build (a Dual Core Processor).
